sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -651,12 +651,12 @@ void CurlSession::ReadStatusLineAndHeadersFromRawResponse(
     bool reuseInternalBuffer)
 {
   auto parser = ResponseBufferParser();
-  auto bufferSize = int64_t();
+  auto bufferSize = int32_t();
```

Comment:
why not `size_t`?

Reply:
The internal buffer used by our libcurl implementation to parse the response from the wire uses an int32_t buffer.
Also, the current logic is to use a negative index to indicate that the internal buffer has been all parsed or does not contain more data from the wire.

updating to use size_t will require to create a new state to control this case.

Also, would you expect to ever be able to pull more than int32_t bytes from the wire in a single read from a socket?  The current implementation tries to pull  only 1024 bytes from the socket for the internal buffer

Reply:
I'm fine with `int32_t`.
The problem is it violates our guideline `DO Represent memory buffer sizes with size_t or ptrdiff_t as appropriate for the environment.`

Reply:
I will ask if we can have an exception to that guideline for when the memory buffer is a well known size and will never grow.